### PR TITLE
Add runners workflow

### PR DESCRIPTION
**Pull request by pavlovic-ivan**
Add workflow that triggers on CI workflow being completed or requested. Add job to deploy runners when CI workflow is requested. Add job to destroy runners when CI workflow is completed

### CODE REVIEW

File: .github/workflows/runners.yaml [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+    workflows: ["CI"]
+    types: 
+      - completed
+      - requested
+
+env:
+  PAT: ${{ secrets.PAT }}
+  GOOGLE_PROJECT: ${{ secrets.GOOGLE_PROJECT }}
+  PULUMI_CONFIG_PASSPHRASE: ${{ secrets.PULUMI_CONFIG_PASSPHRASE }}
+  GOOGLE_CREDENTIALS: ${{ß secrets.GCE_SA_KEY }}
```
Comment (jgiannuzzi) [inline .github/workflows/runners.yaml:14]: looks like a typo 😅
```suggestion
  GOOGLE_CREDENTIALS: ${{ secrets.GCE_SA_KEY }}
```

File: .github/workflows/runners.yaml [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+      - name: Setup Node
+        uses: actions/setup-node@v2
+        with:
+          node-version: '14'
+      - name: Execute Pulumi
+        run: |
+          curl -fsSL https://get.pulumi.com | sh
+          npm install
+          pulumi login $GS_BUCKET
+          pulumi stack init -s $PULUMI_STACK_NAME
```
Comment (jgiannuzzi) [inline .github/workflows/runners.yaml:39]: should this not be the workflow run ID?

**Review by jgiannuzzi — COMMENTED** [review]
Nice work @pavlovic-ivan!

A few questions:
* could you show an example run of this change? (probably would require merging to the default branch first - maybe use a different fork for it?)
* whilst it's fine to have an external repo containing the pulumi code, I believe the ILGPU repo should be the one owning the configuration (at least for image, instance type, and amount of runners). thoughts?

File: .github/workflows/runners.yaml [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+  deploy-runners:
+    runs-on: ubuntu-latest
+    if: ${{ github.event.action == 'requested' }}
+    environment: cuda-test-infra
+    steps:
+      - name: Checkout Pulumi Ephemeral Github Runner project
+        uses: actions/checkout@v2
+        with:
+          repository: pavlovic-ivan/ephemeral-github-runner
+          ref: main
```
Comment (ljubon) [inline .github/workflows/runners.yaml:29]: default branch will be checkout by default, so you should be able to remove this line

File: .github/workflows/runners.yaml [hunk 4]
```diff
@@ -0,0 +1,58 @@
...
+      - name: Execute Pulumi
+        run: |
+          curl -fsSL https://get.pulumi.com | sh
+          npm install
+          pulumi login $GS_BUCKET
+          pulumi stack select $GITHUB_RUN_ID
```
Comment (ljubon) [inline .github/workflows/runners.yaml:57]: as @jgiannuzzi mentioned above, in line 39 you're using `$PULUMI_STACK_NAME` as stack name, and here it's different. 
if I remember correctly, both should `$GITHUB_RUN_ID` from triggered workflow (e.g `gitub.event.workflow_id` ) but also `$GITHUB_RUN_ID` should work fine

File: .github/workflows/runners.yaml (hunk 1 above)
Comment (pavlovic-ivan) [reply .github/workflows/runners.yaml:14]: Thanks for noticing!

File: .github/workflows/runners.yaml (hunk 2 above)
Comment (pavlovic-ivan) [reply .github/workflows/runners.yaml:39]: I have taken Ljubo's advice, and set a the stack to `${{ github.event.workflow.id }}`

File: .github/workflows/runners.yaml (hunk 3 above)
Comment (pavlovic-ivan) [reply .github/workflows/runners.yaml:29]: I was thinking the same, but looks like `ref` is mandatory when checking out a different repo to the one that is running this workflow

File: .github/workflows/runners.yaml (hunk 4 above)
Comment (pavlovic-ivan) [reply .github/workflows/runners.yaml:57]: Thanks, i've taken your advice and set it to `${{ github.event.workflow.id }}`

**Pull request closed without merging**